jupyterhub_traefik_proxy/proxy.py

```diff
@@ -217,21 +231,49 @@ async def _setup_traefik_static_config(self):
         self.static_config = {}
         self.static_config["debug"] = True
         self.static_config["logLevel"] = self.traefik_log_level
+
         entryPoints = {}
+        self.static_config["defaultentrypoints"] = ["http"]
+        entryPoints["http"] = {"address": ":" + str(urlparse(self.public_url).port)}
```

Comment:
this should respect the ip, if given, of public_url. Maybe use `urlparse(self.public_url).netloc`? What happens if public_url has the form `https://1.2.3.4` which has an implicit port of 443 (inferred from https), but the parsed `url.port` will be None.

Reply:
Oh, right! I didn't thought about this case.
Should we just then have one entrypoint `http` or `https` inferred from `public_url`? (I think this is how things happen in CHP)

Right now, (in this form of the PR), there is:
* the http entrypoint - that's setup through `public_url`
* the https entrypoint - that's setup through `traefik_https_port`
And all the requests coming for http are redirected to https.